subsys/bluetooth/host/gatt.c

```diff
@@ -1464,40 +1602,54 @@ static int gatt_indicate(struct bt_conn *conn, u16_t handle,
 	return gatt_send(conn, buf, gatt_indicate_rsp, params, NULL);
 }
 
-#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
-struct sc_data {
-	u16_t start;
-	u16_t end;
-};
-
-static void sc_save(struct bt_gatt_ccc_cfg *cfg,
-		    struct bt_gatt_indicate_params *params)
+#if defined(CONFIG_BT_GATT_SERVICE_CHANGED)
+static void sc_save(u8_t id, bt_addr_le_t *peer, u16_t start, u16_t end)
 {
+	struct gatt_sc_cfg *cfg;
 	struct sc_data data;
 	struct sc_data *stored;
+	bool modified;
 
-	memcpy(&data, params->data, params->len);
+	data.start = sys_le16_to_cpu(start);
+	data.end = sys_le16_to_cpu(end);
```

Comment:
I'd suggest you do the byte-order conversion *before* calling this function (or inline in the function call parameters). Unlike e.g. Linux we don't have byte-order specific integer types like `le16` and `be16` so the risk of loosing track is greater when we pass around values. Because of that I'd recommend that we try to stick to protocol byte order *only* for the values inside packed structs or protocol buffers (byte arrays). Whenever the values are outside of those they should always be in host byte order.

Reply:
I agree, it was just the way it was done before and I wanted to avoid adding too many unrelated changes. I will fix and push again soon!

Reply:
Pushed!